core/shared/src/main/scala/zio/stm/TMap.scala

```diff
@@ -233,6 +233,43 @@ final class TMap[K, V] private (
       TExit.unit
     })
 
+  /**
+   * Removes bindings matching predicate.
+   */
+  def removeIfReturning(p: (K, V) => Boolean): USTM[List[(K, V)]] =
```

Comment:
This should return a `Chunk`.

Reply:
technically, multiple entries could be removed. For instance when you pass the following predicate:

```
for { 
   tmap <- TMap.make("a" -> 1, "aa" -> 2, "aaa" -> 3)
   removed <- tmap.removeIfReturning((_, v) => v > 1)
} yield (removed)
```

Reply:
Sure. A `Chunk` is a collection so if you can return a `List` you can also return a `Chunk`.

Reply:
I changed to use `zio.Chunk` now